pkg/volume/iscsi/iscsi.go

```diff
@@ -0,0 +1,170 @@
+/*
+Copyright 2015 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package iscsi
+
+import (
+	"strconv"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/types"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util/exec"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util/mount"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/volume"
+	"github.com/golang/glog"
+)
+
+// This is the primary entrypoint for volume plugins.
+func ProbeVolumePlugins() []volume.VolumePlugin {
+	return []volume.VolumePlugin{&ISCSIDiskPlugin{nil, exec.New()}}
+}
+
+type ISCSIDiskPlugin struct {
+	host volume.VolumeHost
+	exe  exec.Interface
+}
+
+var _ volume.VolumePlugin = &ISCSIDiskPlugin{}
+
+const (
+	ISCSIDiskPluginName = "kubernetes.io/iscsi"
+)
+
+func (plugin *ISCSIDiskPlugin) Init(host volume.VolumeHost) {
+	plugin.host = host
+}
+
+func (plugin *ISCSIDiskPlugin) Name() string {
+	return ISCSIDiskPluginName
+}
+
+func (plugin *ISCSIDiskPlugin) CanSupport(spec *api.Volume) bool {
+	// see if iscsiadm is there
+	hasAdm := false
+	_, err := plugin.execCommand("iscsiadm", []string{"-h"})
+	if err == nil {
+		hasAdm = true
+	}
+	if spec.ISCSIDisk != nil && hasAdm {
```

Comment:
Check the pointer for nil first and return false if it's nil - don't exec if this is not an ISCSI config.